task_02/roman_kuprii/main.cpp

```diff
@@ -0,0 +1,138 @@
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::to_string;
+using std::vector;
+
+// using namespace std;
+
+struct Edge {
+  int id;
+  std::array<int, 2> nodes;
+
+  Edge(int start, int end, int _id) : id(_id) {
+    nodes[0] = start;
+    nodes[1] = end;
+  }
+
+  std::string to_json() const {
+    std::string res;
+    res = "{ \"id\": ";
+    res += to_string(id);
+    res += ", \"vertex_ids\": [";
+    res += to_string(nodes[0]);
+    res += ", ";
+    res += to_string(nodes[1]);
+    res += "] }";
+    return res;
+  }
+};
+
+// TODO check uniqueness of ID field
+struct Vertex {
+  int id;
+  std::vector<int> edges_ids;
+
+  Vertex(int _id = -1) : id(_id) {}
+
+  std::string to_json() const {
+    std::string res;
+    res = "{ \"id\": ";
+    res += to_string(id) + ", \"edge_ids\": [";
+    for (int n : edges_ids) {
+      res += to_string(n);
+      res += ", ";
+    }
+    res.pop_back();
+    res.pop_back();
+    res += "] }";
+    return res;
+  }
+};
+
+class Graph {
+ public:
+  Graph() : size_(0) {}
+
+  Graph(const vector<Edge>& init_edges, const vector<Vertex>& init_vertices) {
+    vertices_ = init_vertices;
+    edges_ = init_edges;
+    size_ = vertices_.size();
+  }
+
+  std::string to_json() const {
+    std::string res;
+    res = "{ \"vertices\": [ ";
+    for (const auto& v_it : vertices_) {
+      res += v_it.to_json();
+      res += ", ";
+    }
+    res.pop_back();
+    res.pop_back();
+    res += " ], \"edges\": [ ";
+    for (const auto& e_it : edges_) {
+      res += e_it.to_json();
+      res += ", ";
+    }
+    res.pop_back();
+    res.pop_back();
+    res += " ] }\n";
+    return res;
+  }
+
+ private:
+  int size_;
+  vector<Vertex> vertices_;
+  vector<Edge> edges_;
+};
+
+int write_graph(const std::string json_name, Graph A) {
```

Comment:
Не принято передавать значения по константе
```suggestion
int write_graph(std::string json_name, Graph A) {
```

Reply:
Все что сложнее фундаментальных типов, всегда лучше по возможности передавать как `const` ссылка
```suggestion
int write_graph(const std::string& json_name, const Graph& A) {
```